lib/pbench/server/auth/__init__.py

```diff
@@ -27,13 +32,106 @@ class OpenIDCAuthenticationError(OpenIDClientError):
 
 
 class OpenIDClient:
-    """
-    OpenID Connect client object.
-    """
+    """OpenID Connect client object."""
 
     USERINFO_ENDPOINT: Optional[str] = None
     TOKENINFO_ENDPOINT: Optional[str] = None
 
+    class NotConfigured(Exception):
+        """Indicate to the caller a OIDC server is not configured."""
+
+        pass
+
+    class ServerConnectionError(Exception):
+        """Indicate to the caller an unexpected connection error occurred while
+        attempting to connect to the OIDC server.
+        """
+
+        pass
+
+    class ServerConnectionTimedOut(Exception):
+        """Indicate to the caller we timed out attempting to connect to the
+        OIDC server.
+        """
+
+        pass
+
+    @staticmethod
+    def wait_for_oidc_server(
+        server_config: PbenchServerConfig, logger: logging.Logger
+    ) -> str:
+        """Wait for the configured OIDC server to become available.
+
+        Checks if the OIDC server is up and accepting the connections.  The
+        connection check does the GET request on the OIDC server /health
+        endpoint and the sample response returned by the /health endpoint looks
+        like the following:
+
+            {
+                "status": "UP", # if the server is up
+                "checks": []
+            }
+
+        Note: The OIDC server needs to be configured with health-enabled on.
+
+        Args:
+            server_config : the Pbench Server configuration to use
+            logger : the logger to use
+
+        Raises:
+
+            OpenIDClient.NotConfigured : when the given server configuration
+                does not contain the required connection information.
+            OpenIDClient.ServerConnectionError : when any unexpected errors are
+                encountered trying to connect to the OIDC server.
+            OpenIDClient.ServerConnectionTimeOut : when the connection attempt
+                times out.
+        """
+        try:
+            oidc_server = server_config.get("authentication", "server_url")
+        except (NoOptionError, NoSectionError) as exc:
+            raise OpenIDClient.NotConfigured() from exc
+        else:
+            logger.debug("Waiting for OIDC server to become available.")
```

Comment:
I realize you probably just lifted this method from the old location; but that forces a full review. 😆 

Given that the rest of the connection attempt is unindented below rather than nested in the `else:`, it seems like this should be also. Having it structurally separated from the activity it announces makes no sense.